spec/controllers/schools_controller_spec.rb

```diff
@@ -0,0 +1,119 @@
+require 'rails_helper'
+
+RSpec.describe SchoolsController, type: :controller do
+    fixtures :all
+
+    before(:all) do
+        @create_school_name = "University of California, Berkeley"
+        @fail_flash_alert = /Failed to submit information :\(/
+        @success_flash_alert = Regexp.new("Created #{@create_school_name} successfully.")
+    end
+
+    it "allows admin to create" do
+        ApplicationController.any_instance.stub(:require_admin).and_return(true)
+        expect(School.find_by(name: @create_school_name)).to be_nil
+        post :create, {
+            params: { 
+                school: {
+                    name: @create_school_name,
+                    city: "Berkeley",
+                    state: "CA",
+                    website: "www.berkeley.edu"
+                }
+            }
+        }
+        expect(School.find_by(name: @create_school_name)).not_to be_nil
+        assert_match(@success_flash_alert, flash[:success])
+    end
+
+    it "denies teacher to create" do 
+        expect(School.find_by(name: @create_school_name)).to be_nil
+        post :create, {
+            params: {
+                school: {
+                    name: @create_school_name,
+                    city: "Berkeley",
+                    state: "CA",
+                    website: "www.berkeley.edu"
+                }
+            }
+        }
+        expect(School.find_by(name: @create_school_name)).to be_nil
+    end
+
+    it "requires all fields filed" do
```

Comment:
Something really really minor, but 'filled' instead of 'filed' here. Everything looks good!